testplan/report/testing/base.py

```diff
@@ -81,6 +84,20 @@ class Status(object):
         SKIPPED,
     )
 
+    STATUS_PROPAGATION = {
```

Comment:
Can you explain what this is doing exactly? I think this is over-complicating the status precedence mechanism.

Reply:
If a multitest contains a xfail testsuite and the testsuite status is xfail, the multitest status should be passed, not xfail

Reply:
I don't think I agree. If all testcases in a multitest are XFAIL, it is reasonable to mark the multitest as XFAIL.. So long as PASS has higher precedence than XFAIL, if there are a mixture of PASS and XFAIL testcases the MultiTest will be PASS. I don't think this extra complication is required.